contracts/src/periphery/cert/libraries/EigenDACertVerificationLib.sol

```diff
@@ -0,0 +1,390 @@
+    }
+
+    /**
+     * @notice Checks blob inclusion in the batch using Merkle proof
+     * @param batchHeader The batch header
+     * @param blobInclusionInfo The blob inclusion info
+     * @return err Error code (SUCCESS if verification succeeded)
+     * @return errParams Additional error parameters
+     */
+    function checkBlobInclusion(
+        DATypesV2.BatchHeaderV2 memory batchHeader,
+        DATypesV2.BlobInclusionInfo memory blobInclusionInfo
+    ) internal pure returns (StatusCode err, bytes memory errParams) {
+        bytes32 blobCertHash = hashBlobCertificate(blobInclusionInfo.blobCertificate);
+        bytes32 encodedBlobHash = keccak256(abi.encodePacked(blobCertHash));
+        bytes32 rootHash = batchHeader.batchRoot;
+
+        bool isValid = Merkle.verifyInclusionKeccak(
+            blobInclusionInfo.inclusionProof, rootHash, encodedBlobHash, blobInclusionInfo.blobIndex
+        );
+
+        if (isValid) {
+            return (StatusCode.SUCCESS, "");
+        } else {
+            return
+                (StatusCode.INVALID_INCLUSION_PROOF, abi.encode(blobInclusionInfo.blobIndex, encodedBlobHash, rootHash));
+        }
+    }
+
+    /**
+     * @notice Checks the security parameters for a blob cert
+     * @param blobParams The blob params to verify
+     * @param securityThresholds The security thresholds to verify against
+     * @return err Error code (SUCCESS if verification succeeded)
+     * @return errParams Additional error parameters
+     */
+    function checkSecurityParams(
+        DATypesV1.VersionedBlobParams memory blobParams,
+        DATypesV1.SecurityThresholds memory securityThresholds
+    ) internal pure returns (StatusCode err, bytes memory errParams) {
+        uint256 gamma = securityThresholds.confirmationThreshold - securityThresholds.adversaryThreshold;
+        uint256 n = (10000 - ((1_000_000 / gamma) / uint256(blobParams.codingRate))) * uint256(blobParams.numChunks);
+        uint256 minRequired = blobParams.maxNumOperators * 10000;
+
+        if (n >= minRequired) {
+            return (StatusCode.SUCCESS, "");
+        } else {
+            return (StatusCode.SECURITY_ASSUMPTIONS_NOT_MET, abi.encode(gamma, n, minRequired));
+        }
+    }
+
+    /**
+     * @notice Checks quorum signatures and builds a bitmap of confirmed quorums
+     * @param signatureVerifier The signature verifier contract
+     * @param batchHashRoot The hash of the batch header
+     * @param signedQuorumNumbers The signed quorum numbers
+     * @param referenceBlockNumber The reference block number
+     * @param nonSignerStakesAndSignature The non-signer stakes and signature
+     * @param securityThresholds The security thresholds to verify against
+     * @return err Error code (SUCCESS if verification succeeded)
+     * @return errParams Additional error parameters
+     * @return confirmedQuorumsBitmap The bitmap of confirmed quorums
+     */
+    function checkSignaturesAndBuildConfirmedQuorums(
```

Comment:
Why does this say signatureS (with an s)? We're only checking a single (aggregate) signature right?

Reply:
It is the same nomenclature as the underlying signature verifier though.